drivers/lora/sx126x_stm32wl.c

```diff
@@ -45,6 +50,53 @@ void sx126x_dio1_irq_disable(struct sx126x_data *dev_data)
 	irq_disable(DT_INST_IRQN(0));
 }
 
+void sx126x_set_tx_params(int8_t power, RadioRampTimes_t ramp_time)
+{
+	uint8_t buf[2];
+
+	if (pa_output == RFO_LP) {
+		if (power >= 15) {
+			SX126xSetPaConfig(0x07, 0x00, 0x01, 0x01); /* +15 dBm */
+			power = 15;
+		} else if (power <= 10) {
+			SX126xSetPaConfig(0x01, 0x00, 0x01, 0x01); /* +10 dBm */
+		} else {
+			SX126xSetPaConfig(0x04, 0x00, 0x01, 0x01); /* +14 dBm */
+		}
+		if (power < -17) {
+			power = -17;
+		}
+		/* PA overcurrent protection limit 60 mA */
+		SX126xWriteRegister(REG_OCP, 0x18);
+	} else { /* RFO_HP */
+		/* Better Resistance of the RFO High Power Tx to Antenna
+		 * Mismatch, see STM32WL Erratasheet
+		 */
+		SX126xWriteRegister(REG_TX_CLAMP_CFG,
+				    SX126xReadRegister(REG_TX_CLAMP_CFG)
+					| (0x0F << 1));
+		if (power <= 20 && power > 17) {
+			SX126xSetPaConfig(0x03, 0x05, 0x00, 0x01); /* +20 dBm */
+		} else if (power > 14) {
+			SX126xSetPaConfig(0x02, 0x03, 0x00, 0x01); /* +17 dBm */
+		} else if (power <= 14) {
+			SX126xSetPaConfig(0x02, 0x02, 0x00, 0x01); /* +14 dBm */
+		} else {
+			SX126xSetPaConfig(0x04, 0x07, 0x00, 0x01); /* +22 dBm */
+			power = (power > 22) ? 22 : power;
+		}
```

Comment:
This  code seems to be unreachable, given that one of the `power > 14` and `power <= 14` conditions is always true.

What about ordering the comparisons, by decreasing values:
```

if (power >= 20) {
... /* +22 dBm */
} else if (power > 17) {
... /* +20 dBm */
} else if (power > 14) {
... /* +17 dBm */
} else {
... /* +14 dBm */
}
```

Reply:
Good catch, your proposed edit is more readable as well.

However, I realize based on your comment regarding AN5457 that the conditional logic here needs to be reworked. Will address that below.